Adds 4.8 RN bug doc text (#34158)
openshift/openshift-docs: release_notes/ocp-4-8-release-notes.adoc

(link:https://bugzilla.redhat.com/show_bug.cgi?id=1956826[*BZ#1956826*])

* When builds interact with image registries, such as pulling base images, intermittent communications issues can produce build failures. The current release increases the number of retries to these interactions. Now, OpenShift builds are more resilient when they encounter intermittent communication issues with image registries.
(link:https://bugzilla.redhat.com/show_bug.cgi?id=1937535[*BZ#1937535*])

*Cloud Compute*

* Previously, a proxy update caused a full cluster configuration update, including API server restart, during continuous integration (CI) runs. Consequently, some clusters in the Machine API Operator would time out because of unexpected API server outages. This update separates proxy tests and adds postconditions so that clusters in the Machine API Operator become stable again during CI runs. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1913341[*BZ#1913341*])

> * Previously, deleting a machine stuck in `Insufficient disk space on datastore` took longer than expected because there was no distinction between various vCenter task types. With this update, the machine-controller deletion procedure now checks the vCenter task type, and no longer blocks the deletion of the machine-controller. As a result, the machine-controller is deleted quickly.  (link:https://bugzilla.redhat.com/show_bug.cgi?id=1918101[*BZ#1918101*])

Reviewer: ```suggestion
* Previously, deleting a machine stuck in `Insufficient disk space on datastore` took longer than expected because there was no distinction between various vCenter task types. With this update, the machine controller deletion procedure checks the vCenter task type, and no longer blocks the deletion of the machine controller. As a result, the machine controller is deleted quickly.  (link:https://bugzilla.redhat.com/show_bug.cgi?id=1918101[*BZ#1918101*])
```

> * Previously, scaling from zero annotations should not have requeued if the instance type was missing. Consequently, there were constant requeue and error space messages in the `MachineSet controller` logs. With this update, users can set the annotation manually if the instance type is not resolved automatically. As a result, scaling from zero for unknown instance types works so long as users manually provide the annotation. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1918910[*BZ#1918910*])

Reviewer: ```suggestion
* Previously, scaling from zero annotations requeued even if the instance type was missing. Consequently, there were constant requeue and error space messages in the MachineSet controller logs. With this update, users can set the annotation manually if the instance type is not resolved automatically. As a result, scaling from zero for unknown instance types works if users manually provide the annotation. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1918910[*BZ#1918910*])
```

> * Previously, HTTP responses were not closed properly by the machine-api-termination-handler. Consequently, goroutines were leaked in `net.http` read and write loops, which led to high memory usage. This update ensures that HTTP responses are always closed properly. As a result, memory usage is now stable. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1934021[*BZ#1934021*])

Reviewer: ```suggestion
* Previously, HTTP responses were not closed properly by the Machine API termination handler. Consequently, goroutines were leaked in `net.http` read and write loops, which led to high memory usage. This update ensures that HTTP responses are always closed properly. As a result, memory usage is now stable. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1934021[*BZ#1934021*])
```

* Previously, multiple client sets created inside of the MachineSet controller caused slow startup times, which resulted in pods failing readiness checks in some large clusters. Consequently, the MachineSet controller would get stuck in an endless loop. This update fixes the MachineSet controller so that it uses a single client. As a result, the MachineSet controller behaves as expected. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1934216[*BZ#1934216*])

> * Previously, instances would take longer to boot when an upgrade was performed by the Machine Config Daemon on the first boot. Consequently, worker nodes would get stuck in restart loops, and machine healthchecks (MCHs) would remove the worker nodes because they did not properly start. With this update, MHCs no longer remove nodes that have not started correctly. Instead, MHCs only remove nodes when explicitly requested. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1939054[*BZ#1939054*])

Reviewer: ```suggestion
* Previously, instances took longer to boot when an upgrade was performed by the Machine Config Daemon on the first boot. Consequently, worker nodes were stuck in restart loops, and machine healthchecks (MCHs) removed the worker nodes because they did not properly start. With this update, MHCs no longer remove nodes that have not started correctly. Instead, MHCs only remove nodes when explicitly requested. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1939054[*BZ#1939054*])
```

> * Previously, a certificate signing request (CSR) approval was delayed for an unknown reason. Consequently, new machines appearing in the cluster during installation were not approved quick enough, which would prolong cluster installation. This update increases the cache resync period to 10 minutes, instead of 10 hours, to mitigate occasional API server unavailability in early installation phases. As a result, master machines are now approved faster so that cluster installation is no longer prolonged. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1940972[*BZ#1940972*])

Reviewer: ```suggestion
* Previously, a certificate signing request (CSR) approval was delayed for an unknown reason. Consequently, new machines appearing in the cluster during installation were not approved quickly enough, prolonging cluster installation. To mitigate occasional API server unavailability in early installation phases, this update changes the cache resync period from 10 hours to 10 minutes. As a result, control plane machines are now approved faster so that cluster installation is no longer prolonged. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1940972[*BZ#1940972*])
```

> * Previously, the default Google Cloud Platform (GCP) image was out of date and referenced a version from the 4.6 release that did not support newer Ignition versions. Consequently, new machines could not boot in 4.7+ clusters that used the default GCP image. With this update, the GCP image is updated to match the release version. As a result, new machines can now boot with the default GCP image. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1954597[*BZ#1954597*])

Reviewer: ```suggestion
* Previously, the default Google Cloud Platform (GCP) image was out of date and referenced a version from the {product-title} 4.6 release that did not support newer Ignition versions. Consequently, new machines in clusters that used the default GCP image were not able to boot {product-title} 4.7 and later. With this update, the GCP image is updated to match the release version. As a result, new machines can now boot with the default GCP image. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1954597[*BZ#1954597*])
```

> * Previously, due to a strict check of the virtual machine's (VM) 'ProvisioningState' value, the VM could fail during an existence check. With this update, the check is more lenient so that only deleted machines go into a `Failed` phase during an existence check procedure. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1957349[*BZ#1957349*])

Reviewer: ```suggestion
* Previously, due to a strict check of the virtual machine's (VM) `ProvisioningState` value, the VM would sometimes fail during an existence check. With this update, the check is more lenient so that only deleted machines go into a `Failed` phase during an existence check. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1957349[*BZ#1957349*])
```

> * Previously, in an AWS cluster, if you deleted a master machine using `oc delete machine`, the machine was removed from the load balancers. As a result, the load balancer continued to serve requests to removed master machine. With this fix, when you remove a master machine, the load balancer no longer serves requests to the machine.  

Reviewer: ```suggestion
* Previously, if you deleted a control plane machine using `oc delete machine` in an AWS cluster, the machine was removed from the load balancers. As a result, the load balancer continued to serve requests to the removed control plane machine. With this fix, when you remove a control plane machine, the load balancer no longer serves requests to the machine.  
```
